Subsurface/Node/Common/Peripherals/inc/PWMInterface.h

```diff
@@ -0,0 +1,45 @@
+#include "stm32f3xx_hal.h"
+#include <cstdint>
+
+#ifndef PWMINTERFACE_H
+#define PWMINTERFACE_H
+
+class PWMInterface
+{
+public:
+    // Constructor
+    PWMInterface(uint64_t clockSpeed, uint32_t preScaler, uint32_t period,
+    uint32_t pulse, TIM_HandleTypeDef timer, uint32_t channel);
+
+    // Start and stop
+    void PWMStart();
+    void PWMStop();
+
+    // Set parameters
+    void setParameters(uint64_t clockspeed, uint32_t preScaler, uint32_t period,
+    uint32_t pulse, TIM_HandleTypeDef timer, uint32_t channel);
+
+    // Set Prescaler
+    void SET_PRESCALER();
+
+    // Set Autoload
+    void SET_AUTORELOAD();
+
+    // Set Compare
+    void SET_COMPARE();
+
+    // Set Period and Pulse
+    void setPeriod(uint32_t period);
+    void setPulse(uint32_t pulse);
+
+private:
+
+    uint64_t m_clockSpeed;
+    uint32_t m_preScaler;
+    uint32_t m_period;
+    uint32_t m_pulse;
+    uint32_t m_channel;
+    TIM_HandleTypeDef m_timer;
```

Comment:
I think we are gonna want to store a pointer to a `TIM_HandleTypeDef` instead so that we are not copying the object.